actors/market/tests/cron_tick_deal_slashing.rs

```diff
@@ -0,0 +1,398 @@
+            tc.deal_start,
+            tc.deal_end,
+            tc.activation_epoch,
+            SECTOR_EXPIRY,
+        );
+        let deal_proposal = get_deal_proposal(&mut rt, deal_id);
+
+        // terminate
+        rt.set_epoch(tc.termination_epoch);
+        terminate_deals(&mut rt, PROVIDER_ADDR, &[deal_id]);
+
+        // cron tick
+        let cron_tick_epoch = process_epoch(tc.deal_start, deal_id);
+        rt.set_epoch(cron_tick_epoch);
+
+        let (pay, slashed) = cron_tick_and_assert_balances(
+            &mut rt,
+            CLIENT_ADDR,
+            PROVIDER_ADDR,
+            cron_tick_epoch,
+            deal_id,
+        );
+        assert_eq!(tc.payment, pay);
+        assert_eq!(deal_proposal.provider_collateral, slashed);
+        assert_deal_deleted(&mut rt, deal_id, deal_proposal);
+
+        // if there has been no payment, provider will have zero balance and hence should be slashed
+        if tc.payment.is_zero() {
+            assert_account_zero(&mut rt, PROVIDER_ADDR);
+            // client balances should not change
+            let c_locked = get_locked_balance(&mut rt, CLIENT_ADDR);
+            let c_escrow = get_escrow_balance(&rt, &CLIENT_ADDR).unwrap();
+            cron_tick(&mut rt);
+            assert_eq!(c_escrow, get_escrow_balance(&rt, &CLIENT_ADDR).unwrap());
+            assert_eq!(c_locked, get_locked_balance(&mut rt, CLIENT_ADDR));
+        } else {
+            // running cron tick again dosen't do anything
+            cron_tick_no_change(&mut rt, CLIENT_ADDR, PROVIDER_ADDR);
+        }
+        check_state(&rt);
+    }
+}
+
+const START_EPOCH: ChainEpoch = 50;
+const END_EPOCH: ChainEpoch = 50 + 200 * EPOCHS_IN_DAY;
+
+#[test]
+fn deal_is_slashed_at_the_end_epoch_should_not_be_slashed_and_should_be_considered_expired() {
+    let mut rt = setup();
+    let deal_id = publish_and_activate_deal(
+        &mut rt,
+        CLIENT_ADDR,
+        &MinerAddresses::default(),
+        START_EPOCH,
+        END_EPOCH,
+        0,
+        SECTOR_EXPIRY,
+    );
+    let deal_proposal = get_deal_proposal(&mut rt, deal_id);
+
+    // set current epoch to deal end epoch and attempt to slash it -> should not be slashed
+    // as deal is considered to be expired.
+
+    rt.set_epoch(END_EPOCH);
+    terminate_deals(&mut rt, PROVIDER_ADDR, &[deal_id]);
+
+    // on the next cron tick, it will be processed as expired
+    let current = END_EPOCH + 300;
+    rt.set_epoch(current);
+    let (pay, slashed) =
+        cron_tick_and_assert_balances(&mut rt, CLIENT_ADDR, PROVIDER_ADDR, current, deal_id);
+    let duration = END_EPOCH - START_EPOCH;
+    assert_eq!(duration * &deal_proposal.storage_price_per_epoch, pay);
+    assert!(slashed.is_zero());
+
+    // deal should be deleted as it should have expired
+    assert_deal_deleted(&mut rt, deal_id, deal_proposal);
+
+    check_state(&rt);
+}
+
+#[test]
+fn deal_is_correctly_processed_twice_in_the_same_crontick_and_slashed() {
```

Comment:
nit: change bad name from golang tests to match behavior: "deal_payment_and_slashing_correctly_processed_in_same_crontick"